src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs

```diff
@@ -80,6 +80,32 @@ internal static partial class TestWithDynamicLogLevel
             await VerifyAgainstBaselineUsingFile("TestWithDynamicLogLevel.generated.txt", testSourceCode);
         }
 
+        [Fact]
+        public async Task TestBaseline_TestWithNestedClass_Success()
+        {
+            string testSourceCode = @"
+namespace Microsoft.Extensions.Logging.Generators.Tests.TestClasses
+{
+    namespace NestedNamespace
+    {
+        internal static partial class NestedClassTestsExtensions<T1> where T1 : Class1
```

Comment:
Can you also add a test where the outer class isn't `static`?

And also a test for where the outer class isn't `partial`, or at least verify the behavior manually? To ensure the user gets a decent error experience so they know what they did wrong.

Reply:
It might also be good to test with a `struct` outer type as well.

Reply:
> And also a test for where the outer class isn't partial, or at least verify the behavior manually? To ensure the user gets a decent error experience so they know what they did wrong.

For something like
```c#
    class ClassNotPartial
    {
        internal static partial class NestedClass
        {
            [LoggerMessage(EventId = 10, Level = LogLevel.Debug, Message = "M10")]
            public static partial void M10(ILogger logger);
        }
    }
```

We'd get
```
error CS0260: Missing partial modifier on declaration of type 'ClassNotPartial'; another partial declaration of this type exists 
```
I think the error message is clear enough.